components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -72,13 +73,23 @@ class CustomTabWindowFeature(
                     val windowRequest = state.content.windowRequest
                     if (windowRequest?.type == WindowRequest.Type.OPEN) {
                         val intent = configToIntent(state.config)
-                        intent.launchUrl(activity, windowRequest.url.toUri())
-                        store.dispatch(ContentAction.ConsumeWindowRequestAction(sessionId))
+                        // Setting url for intent in order to check if we are able to open that url
+                        intent.intent.data = windowRequest.url.toUri()
+                        if (canHandleIntent(intent)) {
+                            intent.launchUrl(activity, windowRequest.url.toUri())
+                            store.dispatch(ContentAction.ConsumeWindowRequestAction(sessionId))
+                        } else {
+                            handleError()
+                        }
                     }
                 }
         }
     }
 
+    @VisibleForTesting
+    internal fun canHandleIntent(customTabIntent: CustomTabsIntent): Boolean =
+        customTabIntent.intent.resolveActivity(activity.packageManager) != null
```

Comment:
I _think_ `resolveActivity()` will also return activities that may have `exported=false` set and cannot actually be launched? We may need to call `resolveActivity()` on the `PackageManager` and check whether the returned `ActivityInfo` has the exported flag set.